system/jlib/jptree.cpp

```diff
@@ -3046,6 +3064,66 @@ void PTree::serializeToStream(IBufferedSerialOutputStream &tgt) const
     serializeCutOff(tgt, -1, 0);
 }
 
+void PTree::deserializeFromStream(IBufferedSerialInputStream &src)
+{
+    deserializeSelf(src);
+
+    StringBuffer eName;
+    for (;;)
+    {
+        eName.clear();
+        peekReadZeroTerminatedString(src, eName);
```

Comment:
It was not clear to me that this copied the string, but did not advance.  Clearer to have

const char * nextName = queryZeroTerminatedString(src, len);
if (len == 0)
   src.skip(1);

//Save the name, but do not consume the input
eName.append(nextName);